azurelinuxagent/common/cgroupstelemetry.py

```diff
@@ -51,18 +51,19 @@ def track_cgroup(cgroup):
         Adds the given item to the dictionary of tracked cgroups
         """
         with CGroupsTelemetry._rlock:
-            CGroupsTelemetry._tracked.append(cgroup)
-            logger.info("Started tracking new cgroup: {0}, path: {1}".format(cgroup.name, cgroup.path))
+            if not CGroupsTelemetry.is_tracked(cgroup.name, cgroup.controller, cgroup.path):
+                CGroupsTelemetry._tracked.append(cgroup)
+                logger.info("Started tracking new cgroup: {0}, path: {1}".format(cgroup.name, cgroup.path))
 
     @staticmethod
-    def is_tracked(name, controller):
+    def is_tracked(name, controller, path):
```

Comment:
why do we need the path on top of the name and controller?

Reply:
Discussed offline. Will change this.